homeassistant/components/airvisual_pro/__init__.py

```diff
@@ -24,19 +33,41 @@
 UPDATE_INTERVAL = timedelta(minutes=1)
 
 
+@dataclass
+class AirVisualProData:
+    """Define a data class."""
+
+    coordinator: DataUpdateCoordinator
+    node: NodeSamba
+
+
 async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     """Set up AirVisual Pro from a config entry."""
+    node = NodeSamba(entry.data[CONF_IP_ADDRESS], entry.data[CONF_PASSWORD])
+
+    try:
+        await node.async_connect()
+    except NodeProError as err:
+        raise ConfigEntryNotReady() from err
+
+    reload_task: asyncio.Task | None = None
 
     async def async_get_data() -> dict[str, Any]:
         """Get data from the device."""
         try:
-            async with NodeSamba(
-                entry.data[CONF_IP_ADDRESS], entry.data[CONF_PASSWORD]
-            ) as node:
-                return await node.async_get_latest_measurements()
+            data = await node.async_get_latest_measurements()
+        except NodeConnectionError as err:
+            nonlocal reload_task
+            if not reload_task:
```

Comment:
Why do we need this check? There is only one coordinator per config entry, right? Isn't it very unlikely that the coordinator would have time to update again before the task unloads the config entry?

We never reset the `reload_task` to `None` it seems.